File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/IUserInfoRepository.java
package ru.usedesk.chat_sdk.external;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import ru.usedesk.sdk.external.entity.chat.UsedeskConfiguration;
import ru.usedesk.sdk.external.entity.exceptions.DataNotFoundException;

public interface IUserInfoRepository {

    @NonNull
    String getToken() throws DataNotFoundException;

    void setToken(@Nullable String token);

    @NonNull
    UsedeskConfiguration getConfiguration() throws DataNotFoundException;

    void setConfiguration(@Nullable UsedeskConfiguration configuration);
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/repository/configuration/IUserInfoRepository.kt

package ru.usedesk.chat_sdk.data.repository.configuration

import ru.usedesk.chat_sdk.entity.UsedeskChatConfiguration

internal interface IUserInfoRepository {
    fun getConfiguration(): UsedeskChatConfiguration?

    fun updateConfiguration(onUpdate: UsedeskChatConfiguration.() -> UsedeskChatConfiguration)
}